content/general/concepts/search-algorithms/search-algorithms.md

```diff
@@ -0,0 +1,64 @@
+---
+Title: 'Search Algorithms'
+Description: 'Search algorithms are a class of algorithms used to find an optimal solution to a problem by exploring a search space.'
+Subjects:
+  - 'Machine Learning'
+  - 'Computer Science'
+Tags:
+  - 'AI'
+  - 'Algorithms'
+  - 'Graphs'
+  - 'Graph Search'
+CatalogContent:
+  - 'paths/data-science'
+---
+
+A **search algorithm** is a type of algorithm used in artificial intelligence to find the best or most optimal solution to a problem by exploring a set of possible solutions, also called a *search space*. A search algorithm filters through a large number of possibilities to find a solution that works best for a given set of constraints.
+
+Search algorithms typically operate by organizing the search space into a particular type of graph, commonly a tree, and evaluate the best score, or *cost*, of traversing each branch of the tree. A solution is a path from the start state to the goal state that optimizes the cost given the parameters of the implemented algorithm.
+
+Search algorithms are typically organized into two categories:
+
+ - Uninformed Search
+
+    Uninformed search algorithms consist of general purpose traversals of the state space or search tree without any information about how good a state is. These are also referred to as blind search algorithms.
+
+ - Informed Search
```

Comment:
```suggestion
Informed Search: algorithms that have information about the goal during the traversal, allowing the search to prioritize its expansion toward the goal state instead of exploring directions that may yield a favorable cost but don't lead to the goal, or global optimum. By including extra rules that aid in estimating the location of the goal (known as _heuristics_) informed search algorithms can be more computationally efficient when searching for a path to the goal state.
```
```